### PR TITLE
MINOR: [Parquet][Docs] Update note on parquet RLE support

### PR DESCRIPTION
**Pull request by milesgranger**
Boolean values supported now,  ref [ARROW-17450](https://issues.apache.org/jira/browse/ARROW-17450) and [ARROW-18031](https://issues.apache.org/jira/browse/ARROW-18031)

### CODE REVIEW

**Comment by jorisvandenbossche** [comment]
I would maybe keep a note, but update it to mention that it is supported _only_ for boolean type in case of values?

File: docs/source/cpp/parquet.rst [hunk 1]
```diff
@@ -102,7 +102,8 @@ Encodings
 | DELTA_LENGTH_BYTE_ARRAY  | ✓        |          |         |
 +--------------------------+----------+----------+---------+
 
-* \(1) Only supported for encoding definition and repetition levels, not values.
+* \(1) Only supported for encoding definition, repetition levels,
```
Comment (jorisvandenbossche) [inline docs/source/cpp/parquet.rst:105]:
```suggestion
* \(1) Only supported for encoding definition and repetition levels,
```

(it's also "definition level", so I think it requires an "and" here to read correctly?)

Comment (milesgranger) [reply docs/source/cpp/parquet.rst:105]: ah, okay. I went back and forth on that one, if those belonged together. Thank you.

**Review by jorisvandenbossche — APPROVED** [review]
Thanks!

**Pull request merged**